packages/dashboard/src/components/tasks/tasks-app.tsx

```diff
@@ -270,23 +290,29 @@ export const TasksApp = React.memo(
           return {};
         }
 
-        const taskIds: string[] = tasks.map((task) => task.booking.id);
-        const taskIdsQuery = taskIds.join(',');
-        const taskRequests = (await rmf.tasksApi.queryTaskRequestsTasksRequestsGet(taskIdsQuery))
-          .data;
-
         const taskRequestMap: Record<string, TaskRequest> = {};
-        let requestIndex = 0;
-        for (const id of taskIds) {
-          if (requestIndex < taskRequests.length && taskRequests[requestIndex]) {
-            taskRequestMap[id] = taskRequests[requestIndex];
+        const allTaskIds: string[] = tasks.map((task) => task.booking.id);
+        const queriesRequired = Math.ceil(allTaskIds.length / QueryLimit);
+        for (let i = 0; i < queriesRequired; i++) {
```

Comment:
We don't need `queriesRequired` (and the task count endpoint), the frontend can just queries until there is no more result.

A potential use of the task count is to provide a progress gauge but I don't think we are doing it, but if so, we can try using the task date to give an estimated progress.

Reply:
https://github.com/open-rmf/rmf-web/pull/909/commits/1de251a91f11ab970c2f114b16c69fe514ff7b63

Reply:
`exportTasksToCsv` still fetches all tasks, it is also doing unneeded double query, first it query for the task states then it query or the task requests, it could just query the task requests directly.

Reply:
This is a workaround until we get pickup and destination information into labels, basically using https://github.com/open-rmf/rmf_ros2/pull/328 (we had this conversation over VC to use labels for warning time and other pieces of information)

We are currently using task requests to parse for pickup and destination information as it is the only source available, since phases change over time. Task states also do not contain it's request.

Not all task requests are getting queried in `getPastMonthTaskRequests` too, just the requests of the task states that we got from `getPastMonthTasks`, and also queried in chunks of 100.

Added a fixme to make the intention more clear, https://github.com/open-rmf/rmf-web/pull/909/commits/d1267670d119315eecfd86fed18ecb5b81b449c4